main.cpp

```diff
@@ -0,0 +1,284 @@
+	while(opcion < 0){
+		cout<<"Ingrese la opci�n deseada: ";
+		cin>>opcion;
+		if( (opcion == 0 && !permiteSalir) || opcion < 0 || opcion > opciones ){
+			cout<<"La opci�n ingresada no es v�lida"<<endl;
+			opcion = NOT_FOUND;
+		}
+	}
+	return opcion;
+}
+
+/*
+	Ejecuta la opci�n elegida. 
+*/
+void EjecutarOpcion(int opcion){
+	switch(opcion){
+		case 1:
+			AgregarEquipo();
+			MostrarMenu(false);			
+			break;
+		case 2:
+			EliminarEquipo();
+			MostrarMenu(false);			
+				break;
+		case 3:
+			ListarEquipos();
+			MostrarMenu(false);
+			break;				
+	}
+}
+
+
+/*
+	Imprime en pantalla la lista de los equipos anotados (en el archivo).
+*/
+void ListarEquipos(){
+	if(CantidadEquipos() == 0){
+		cout<<"No existen equipos anotados";
+	}else{
+		archivo = fopen(NOMBRE_ARCHIVO,"rb");
+		Equipo equipo;
+		cout<<"Lista de Equipos:"<<endl;
+		cout << left << setw (5) << "Id" << setw (33) << "Nombre" << setw (18) << "(Ataque - Defensa)" << endl;
+		fread(&equipo, sizeof(equipo), 1, archivo);
+		while(!feof(archivo)){
+			cout << left << setw (5) << equipo.id << setw (33) << equipo.nombre << "("<< equipo.potenciaAtaque<<" - "<<equipo.potenciaDefensa<<")"<< endl ;
+			fread(&equipo, sizeof(equipo), 1, archivo);
+		}
+		fclose(archivo);
+	}
+}
+
+Equipo CrearEquipo(bool permiteNuevo){
+	int pos = NOT_FOUND;
+	char modificarEquipo;	
+	Equipo equipo;
+
+	// Si el id existe, pregunta si desea modificar, si no te permite cargar un nuevo id
+	do{
+		cout<<"Ingrese identificador: ";
+		cin>>equipo.id;
+		pos = PosicionDelEquipoEnArchivo(equipo.id);
+		if(pos > NOT_FOUND){
+			cout<<"El equipo ya se encuentra ingresado. �Desea modificarlo? (s/n)";
+			cin>>modificarEquipo;
+		}		
+	} while(pos > NOT_FOUND && modificarEquipo == 'n');
+	
+	if(pos > NOT_FOUND && !permiteNuevo){
+		cout<<"Ha alcanzado el l�mite m�ximo de equipos. Para ingresar m�s, deber� eliminar alguno. ";
+	}
+	
+	cout<<"Ingrese nombre: ";
+	cin>>equipo.nombre;
+	cout<<"Ingrese potencia en ataque: ";
+	cin>>equipo.potenciaAtaque;
+	cout<<"Ingrese potencia en defensa: ";
+	cin>>equipo.potenciaDefensa;
+	return equipo;
+}
+
+void AgregarEquipoAlArchivo(Equipo equipo){
+	archivo = fopen(NOMBRE_ARCHIVO,"ab");
+	fseek(archivo, 0L, SEEK_END);
+	fwrite(&equipo, sizeof(Equipo), 1, archivo);
+	cout<<"Equipo agregado con �xito"<<endl;
+	fclose(archivo);
+}
+
+void ReescribirArchivo(Equipo equipos[], int tamanio){
+	archivo = fopen(NOMBRE_ARCHIVO, "wb");	
+	fwrite(&equipos, sizeof(Equipo), tamanio, archivo);	
+	fclose(archivo);
+}
+
+/*
+	Agrega o modifica un equipo.
+*/
+void AgregarEquipo(){
+	Equipo equipo = CrearEquipo(CantidadEquipos() < 100);
+	int pos = PosicionDelEquipoEnArchivo(equipo.id);	
+	if(pos < 0){
+		AgregarEquipoAlArchivo(equipo);	
+	}else{			
+		int tamanio = CantidadEquipos();
+		Equipo equipos[tamanio];
+		SetearListaDeEquipos(equipos);
+		equipos[pos] = equipo;		
+		
+		archivo = fopen(NOMBRE_ARCHIVO, "wb");	
+		fwrite(&equipos, sizeof(Equipo), tamanio, archivo);	
```

Comment:
Gracias por las correcciones! Después vemos de corregirlas!
Saludos!

José María Cullen
________________________________
De: José María Cullen<mailto:jose_cullen@hotmail.com>
Enviado el: ‎28/‎06/‎2015 22:01
Para: josecullen/algoritmos-utn<mailto:reply@reply.github.com>; josecullen/algoritmos-utn<mailto:algoritmos-utn@noreply.github.com>
Asunto: RE: [algoritmos-utn] Correcciones (#1)

Esto intentamos hacer.. Se rompía el archivo 📁... Fue en lo que más tiempo perdimos... Algo nos debe haber faltado pero podamos bastante.
En clase por ahí lo vemos si te parece.
Igual con el cambio de cargar al principio en listas esto va a ser más fácil

José María Cullen
________________________________
De: Emanuel<mailto:notifications@github.com>
Enviado el: ‎28/‎06/‎2015 17:38
Para: josecullen/algoritmos-utn<mailto:algoritmos-utn@noreply.github.com>
Asunto: Re: [algoritmos-utn] Correcciones (#1)

> +/*
> +     Agrega o modifica un equipo.
> +*/
> +void AgregarEquipo(){
> +     Equipo equipo = CrearEquipo(CantidadEquipos() < 100);
> +     int pos = PosicionDelEquipoEnArchivo(equipo.id);
> +     if(pos < 0){
> +             AgregarEquipoAlArchivo(equipo);
> +     }else{
> +             int tamanio = CantidadEquipos();
> +             Equipo equipos[tamanio];
> +             SetearListaDeEquipos(equipos);
> +             equipos[pos] = equipo;
> +
> +             archivo = fopen(NOMBRE_ARCHIVO, "wb");
> +             fwrite(&equipos, sizeof(Equipo), tamanio, archivo);

Para modificar un solo registro pueden directamente pararse en el lugar que corresponda y pisar un solo registro, algo así:
```c++
fseek(archivo, sizeof(Equipo) * pos, SEEK_BEG);
fwrite(&equipo, sizeof(Equipo), 1, archivo);
```

---
Reply to this email directly or view it on GitHub:
https://github.com/josecullen/algoritmos-utn/pull/1/files#r33426649